### PR TITLE
add a callstack to errors in `browser`, `API`, `msg`

### PR DESCRIPTION
**Pull request by tophf**
Testing in devtools console:
* `API` callstack: `await API.styles.delete(1e9)` in the style manager.
* `msg.send` callstack: `await API.styles.delete(1e9)` in any web site after choosing Stylus context in the console toolbar.
* `browser` callstack: `await browser.tabs.get(1e9)` in the style manager.

@eight04, WDYT?

### CODE REVIEW

File: js/msg.js [hunk 1]
```diff
@@ -154,18 +160,23 @@
...
       const message = {method: 'invokeAPI', path, args};
-      let res;
       // content scripts, probably private tabs, and our extension tab during Chrome startup
       if (!bg) {
-        res = msg.send(message);
-      } else {
-        res = deepMerge(await bg.msg._execute(TARGETS.extension, message, {
+        return msg.send(message);
+      }
+      const err = new Error(`Callstack before invoking API.${path.join('.')}:`);
```
Comment (eight04) [inline js/msg.js:169]: It seems that we don't need this according to my screenshot?

**Review by eight04 — COMMENTED** [review]
Tested with:
```diff
diff --git a/manage.html b/manage.html
index c0eddb19..f79ed4cc 100644
--- a/manage.html
+++ b/manage.html
@@ -370,6 +370,6 @@
     <path d="M13.3,12.8l1.5-2.6l-2.2-1.5c0-0.2,0.1-0.5,0.1-0.7c0-0.2,0-0.5-0.1-0.7l2.2-1.5l-1.5-2.6l-2.4,1.2 c-0.4-0.3-0.8-0.5-1.2-0.7L9.5,1h-3L6.3,3.7C5.9,3.8,5.5,4.1,5.1,4.4L2.7,3.2L1.2,5.8l2.2,1.5c0,0.2-0.1,0.5-0.1,0.7 c0,0.2,0,0.5,0.1,0.7l-2.2,1.5l1.5,2.6l2.4-1.2c0.4,0.3,0.8,0.5,1.2,0.7L6.5,15h3l0.2-2.7c0.4-0.2,0.8-0.4,1.2-0.7L13.3,12.8z  M8,10.3c-1.3,0-2.3-1-2.3-2.3c0-1.3,1-2.3,2.3-2.3c1.3,0,2.3,1,2.3,2.3C10.3,9.3,9.3,10.3,8,10.3z"/>
   </symbol>
 </svg>
-
+<script src="test.js"></script>
 </body>
 </html>
diff --git a/manifest.json b/manifest.json
index 4ed8a4a1..74ad5a6b 100644
--- a/manifest.json
+++ b/manifest.json
@@ -77,7 +77,8 @@
         "js/msg.js",
         "js/prefs.js",
         "content/style-injector.js",
-        "content/apply.js"
+        "content/apply.js",
+        "test.js"
       ]
     },
     {
diff --git a/test.js b/test.js
new file mode 100644
index 00000000..9a042567
--- /dev/null
+++ b/test.js
@@ -0,0 +1,3 @@
+API.styles.delete(1e9);
+browser.tabs.get(1e9);
+

```
*FF before/after the patch (the patch has no effect. Actually I think the stack is changed but FF doesn't display them in the default error reporter?)*
image
image
*Chrome before patch*
image
image
*Chrome after patch*
image
image

**Comment by eight04** [comment]
Yeah it actually works (except the polyfill of course):
```diff
diff --git a/manage.html b/manage.html
index c0eddb19..f79ed4cc 100644
--- a/manage.html
+++ b/manage.html
@@ -370,6 +370,6 @@
     <path d="M13.3,12.8l1.5-2.6l-2.2-1.5c0-0.2,0.1-0.5,0.1-0.7c0-0.2,0-0.5-0.1-0.7l2.2-1.5l-1.5-2.6l-2.4,1.2 c-0.4-0.3-0.8-0.5-1.2-0.7L9.5,1h-3L6.3,3.7C5.9,3.8,5.5,4.1,5.1,4.4L2.7,3.2L1.2,5.8l2.2,1.5c0,0.2-0.1,0.5-0.1,0.7 c0,0.2,0,0.5,0.1,0.7l-2.2,1.5l1.5,2.6l2.4-1.2c0.4,0.3,0.8,0.5,1.2,0.7L6.5,15h3l0.2-2.7c0.4-0.2,0.8-0.4,1.2-0.7L13.3,12.8z  M8,10.3c-1.3,0-2.3-1-2.3-2.3c0-1.3,1-2.3,2.3-2.3c1.3,0,2.3,1,2.3,2.3C10.3,9.3,9.3,10.3,8,10.3z"/>
   </symbol>
 </svg>
-
+<script src="test.js"></script>
 </body>
 </html>
diff --git a/manifest.json b/manifest.json
index 4ed8a4a1..74ad5a6b 100644
--- a/manifest.json
+++ b/manifest.json
@@ -77,7 +77,8 @@
         "js/msg.js",
         "js/prefs.js",
         "content/style-injector.js",
-        "content/apply.js"
+        "content/apply.js",
+        "test.js"
       ]
     },
     {
diff --git a/test.js b/test.js
new file mode 100644
index 00000000..cdb9c7fd
--- /dev/null
+++ b/test.js
@@ -0,0 +1,7 @@
+function myLog(err) {
+  console.error(err);
+  console.log(err.stack);
+}
+API.styles.delete(1e9).catch(myLog);
+browser.tabs.get(1e9).catch(myLog);
+
```
image
image

File: js/msg.js (hunk 1 above)
Comment (tophf) [reply js/msg.js:169]: Your screenshot shows a local callstack after `Callstack before invoking API.styles.delete`, which is the entire point of this PR, so we need it.

Comment (eight04) [reply js/msg.js:169]: Before the patch, it already shows the correct stack:
image
After the patch, it shows the same stack twice:
image

**Pull request merged**